provider/pdns/pdns.go

```diff
@@ -394,7 +392,7 @@ func (p *PDNSProvider) mutateRecords(endpoints []*endpoint.Endpoint, changetype
 	for _, zone := range zonelist {
 		jso, err := json.Marshal(zone)
 		if err != nil {
-			log.Errorf("JSON Marshal for zone struct failed!")
+			return provider.NewSoftError(fmt.Errorf("JSON Marshal for zone struct failed: %v", err))
```

Comment:
What makes you think it should be a soft error, here ?
At first glance, it seems it should be a regular error.

Reply:
Yes, I think you are correct!